library/core/src/sync/atomic.rs

```diff
@@ -3402,11 +3402,8 @@ macro_rules! atomic_int {
             /// This method is mostly useful for FFI, where the function signature may use
             #[doc = concat!("`*mut ", stringify!($int_type), "` instead of `&", stringify!($atomic_type), "`.")]
             ///
-            /// Returning an `*mut` pointer from a shared reference to this atomic is safe because the
-            /// atomic types work with interior mutability. All modifications of an atomic change the value
-            /// through a shared reference, and can do so safely as long as they use atomic operations. Any
-            /// use of the returned raw pointer requires an `unsafe` block and still has to uphold the same
-            /// restriction: operations on it must be atomic.
+            /// All modifications of an atomic change the value through a shared reference, and can do so safely
+            /// as long as they use atomic operations.
```

Comment:
Why did you remove the first sentence of this paragraph? I think just this sentence on its own is quite confusing.

It may also be good to remind the caller that any use of the returned raw pointer is subject to the rules described in the module-level docs, linking to https://doc.rust-lang.org/nightly/std/sync/atomic/index.html#memory-model-for-atomic-accesses.